MacrobenchmarkSample/ftl/README.md

```diff
@@ -0,0 +1,292 @@
+
+1. Github Workflow
+  - Builds the app (under test) and the macrobenchmarks.
+  - Submits the app APK, and the macrobenchmark APK to Firebase Test Lab.
+  
+2. A Google Cloud function which automatically pulls results from Firebase Test Lab, and creates custom metrics using Google Cloud Monitoring.
+
+3. Google Cloud Monitoring dashboards
+  - Set up custom dashboards
+  - Set up alerts
+
+## Project Setup
+
+```
+You will need a Google Account to setup your Google Cloud Platform project.
+```
+
+If you don't have a Firebase project for your app, go to the [Firebase Console](https://console.firebase.google.com/u/0/?pli=1) and click `Create New Project` to create one now. You will need ownership or edit permissions in your project. You might also want to upgrade to the [Blaze plan](https://firebase.google.com/docs/test-lab/usage-quotas-pricing), if you run out of [free quota](https://firebase.google.com/docs/test-lab/usage-quotas-pricing).
+
+### One Time Setup
+
+#### Create Secrets
+
+Create a Google Cloud IAM [service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys). This allows your GitHub project to authenticate with the Google Cloud Platform. Store this service account key using [GitHub Secrets](https://docs.github.com/en/actions/reference/encrypted-secrets). 
+
+**Do not** check in this key to your repository, or share with anyone outside your organization.
+
+Create 2 secrets:
+
+* `GCP_PROJECT_ID` represents your Google Cloud Platform `project id`. 
+* `GCP_SA_KEY` represents the service account key exported to JSON.
+
+Now that you have a service account, you also need to grant the service account access to the Firebase Test Lab APIs. For more information, please refer to [enabling an API for your project](https://cloud.google.com/endpoints/docs/openapi/enable-api).
+
+#### Create a Cloud Storage Bucket
+
+[Create](https://cloud.google.com/storage/docs/creating-buckets) a Google Cloud Storage bucket in your project called `macrobenchmark-results`. This will be the location where the results of test runs from Firebase Test Lab are stored. 
+
+#### Deploy the Google Cloud Function
+
+A [Google Cloud function](#TODO) is used to process the results of test runs from Firebase Test Lab, and process the Macrobenchmark results `json` file. The cloud function also captures all the quantitative metrics from the `json` file, and uploads them to Google Cloud Monitoring.
+
+The cloud function uses the [default service account](https://cloud.google.com/functions/docs/securing/function-identity) associated with the Google Cloud Platform project. This service account should be assigned roles that give it access to Google Cloud Storage (so it has access to the created buckets and results stored by Firebase Test Lab), and Google Cloud Monitoring. For more information, please refer to [enabling an API for your project](https://cloud.google.com/endpoints/docs/openapi/enable-api).
```

Comment:
unrelated to this PR but technically, the github action could do this as well, no? (pull the result and expert data into monitoring?)

Reply:
That is correct. We originally wanted FTL team to own this and provide this as a service. But they declined. We could simplify this flow in the future, and make this an action.